packages/core/src/RenderingEngine/Viewport.ts

```diff
@@ -527,6 +532,78 @@ class Viewport implements IViewport {
     return intersections;
   }
 
+  /**
+   * Sets the camera to an initial bounds. If
+   * resetPan and resetZoom are true it places the focal point at the center of
+   * the volume (or slice); otherwise, only the camera zoom and camera Pan or Zoom
+   * is reset for the current view.
+   * @param displayArea - The display area of interest.
+   * @param suppressEvents - If true, don't fire displayArea event.
+   */
+  public setDisplayArea(
+    displayArea: DisplayArea,
+    suppressEvents = false
+  ): void {
+    const storeAsInitialCamera = displayArea.storeAsInitialCamera;
+    this.setCamera(this.fitToCanvasCamera, false);
+
+    const { imageArea, imageCanvasPoint } = displayArea;
+
+    if (imageArea) {
+      const { areaX, areaY } = imageArea;
+      const zoom = Math.min(this.getZoom() / areaX, this.getZoom() / areaY);
```

Comment:
The correct calculation for this is:
zoomWidth = areaX * canvas.width / image.width
zoomHeight = areaY * canvas.width / image.height
setZoom( getZoom() / min(zoomWidth,zoomHeight) ) 
assuming zoom fits the larger of the width and height.

Reply:
So what I do is I treat the fitToCanvas camera as the zoom=1 case for the DisplayArea API. Then I increase or decrease the factor relative to the fitToCanvas camera. The actual zoom stored is still absolute (since the fitToCanvas camera calculates absolute camera). However, the DisplayArea API is all relative to what "fitToCanvas" is defined by, not an absolute coordinate system.